pyserini/trectools/_base.py

```diff
@@ -90,36 +116,45 @@ def get_all_topics_from_runs(runs) -> Set[str]:
         return all_topics
 
     @staticmethod
-    def merge(runs, aggregation: str, depth: int = None, k: int = None):
+    def merge(runs, aggregation: AggregationMethod, depth: int = None, k: int = None):
         if len(runs) < 2:
             raise Exception('Merge requires at least 2 runs.')
 
         rows = []
 
-        if aggregation == 'sum':
+        if aggregation == AggregationMethod.SUM:
             for topic in TrecRun.get_all_topics_from_runs(runs):
                 doc_scores = dict()
                 for run in runs:
                     for topic, _, docid, _, score, _ in run.get_docs_by_topic(topic, depth).to_numpy():
                         doc_scores[docid] = doc_scores.get(docid, 0.0) + score
 
                 sorted_doc_scores = sorted(iter(doc_scores.items()), key=lambda x: (-x[1], x[0]))
-                sorted_doc_scores = sorted_doc_scores if k is None else sorted_doc_scores[:k] 
+                sorted_doc_scores = sorted_doc_scores if k is None else sorted_doc_scores[:k]
 
                 for rank, (docid, score) in enumerate(sorted_doc_scores, start=1):
                     rows.append((topic, 'Q0', docid, rank, score, 'merge_sum'))
 
         else:
-            raise Exception(f'Unknown aggregation method {aggregation} detected.')
+            raise NotImplementedError()
 
-        return TrecRun.get_trec_run_by_list(rows)
+        return TrecRun.get_trec_run_from_list(rows)
 
     @staticmethod
-    def get_trec_run_by_list(rows, run=None):
+    def get_trec_run_from_list(rows, run: TrecRun = None) -> TrecRun:
         res = TrecRun() if run is None else run
 
         df = pd.DataFrame(rows)
         df.columns = TrecRun.columns
         res.run_data = df.copy()
 
         return res
+
+    @staticmethod
+    def generate_trec_run_from_search_results(hits: List[JSimpleSearcherResult]) -> TrecRun:
```

Comment:
how about just `from_search_results`? The fact that it's a static of `TrecRun` makes the return type obvious?